### PR TITLE
Remove zk

### PR DESCRIPTION
**Pull request by lispc**

### CODE REVIEW

File: halo2_proofs/src/plonk/prover.rs [hunk 1]
```diff
@@ -490,6 +494,7 @@ pub fn create_proof<
         .collect::<Result<Vec<_>, _>>()?;
 
     // Commit to the vanishing argument's random polynomial for blinding h(x_3)
+    #[cfg(feature = "zero-knowledge")]
```
Comment (lispc) [inline halo2_proofs/src/plonk/prover.rs:497]: why

File: halo2_proofs/src/plonk/lookup/prover.rs [hunk 2]
```diff
@@ -614,9 +614,20 @@ fn permute_expression_pair<C: CurveAffine, R: RngCore>(
...
     assert!(repeated_input_rows.is_empty());
 
-    permuted_input_expression
-        .extend((0..(blinding_factors + 1)).map(|_| C::Scalar::random(&mut rng)));
-    permuted_table_coeffs.extend((0..(blinding_factors + 1)).map(|_| C::Scalar::random(&mut rng)));
+    permuted_input_expression.extend((0..(blinding_factors + 1)).map(|_| {
+        if cfg!(feature = "zero-knowledge") {
+            C::Scalar::random(&mut rng)
+        } else {
+            C::Scalar::zero()
```
Comment (lispc) [inline halo2_proofs/src/plonk/lookup/prover.rs]: in fact if zk is disabled, "0..(blinding_factors + 1" becomes empty so "C::Scalar::zero()" never executed.

what about : 

```
#[cfg(feature = "zero-knowledge")]
permuted_table_coeffs.extend((0..(blinding_factors + 1)).map(|_| C::Scalar::random(&mut rng)));
```

File: halo2_proofs/src/plonk/vanishing/prover.rs [hunk 3]
```diff
@@ -107,6 +112,57 @@ impl<C: CurveAffine> Committed<C> {
     }
 }
 
+#[cfg(not(feature = "zero-knowledge"))]
```
Comment (lispc) [inline halo2_proofs/src/plonk/vanishing/prover.rs]: can we use less code ( i mean reuse existing codes as much as possible ) here?

the less code, the less maintain burden

**Comment by lispc** [comment]
1. a goal of removing zk is that we can use every row(2\*\*degree) in circuit.  So maybe it is better to add a test, to demonstrate every row of a 2\*\*k circuit can be assigned and MockProver::verify-ed.
2. there are many "feature(zero-knowlegde)" in this PR, can we use less? the more these "if cfg.." , the more long term code maintenance burden

**Comment by spherel** [comment]
### Summary: 
To remove zero-knowledge, we need to 
1. change the `blinding_factors` to be 0. So the column structure should be `[usable_rows, last_row]`.
2. remove randomness from the end of each witness column and expression (but the last row is still out of usable rows so sometimes padding a `zero` is needed).
3. remove the `random_poly` and its commitment in the vanishing argument. 

`examples/simple-example-4.rs` constructs a circuit with `2^k - 1` rows.

**Pull request closed without merging**